unpackaged/functional_tests/address/classes/Address_FTST.cls

```diff
@@ -380,4 +380,397 @@ public with sharing class Address_FTST {
+        System.assertEquals(2, newAccountAddressList.size(), 'There should be exactly 2 addresses inserted');
+        System.assertEquals(state, newAccountAddressList[0].hed__MailingState__c, 'The state should be ' + state);
+        System.assertEquals(city, newAccountAddressList[0].hed__MailingCity__c, 'The city should be ' + city);
+        System.assertEquals(street, newAccountAddressList[0].hed__MailingStreet__c, 'The street should be ' + street);
+        System.assertEquals(county, newAccountAddressList[0].hed__MailingCounty__c, 'The county should be ' + county);
+        System.assertEquals(
+            country,
+            newAccountAddressList[0].hed__MailingCountry__c,
+            'The country should be ' + country
+        );
+        System.assertEquals(
+            postalCode,
+            newAccountAddressList[0].hed__MailingPostalCode__c,
+            'The postal code should be ' + postalCode
+        );
+        System.assertEquals(state, newAccountAddressList[1].hed__MailingState__c, 'The state should be ' + state);
+        System.assertEquals(newCity, newAccountAddressList[1].hed__MailingCity__c, 'The city should be ' + newCity);
+        System.assertEquals(street, newAccountAddressList[1].hed__MailingStreet__c, 'The street should be ' + street);
+        System.assertEquals(county, newAccountAddressList[1].hed__MailingCounty__c, 'The county should be ' + county);
+        System.assertEquals(
+            newCountry,
+            newAccountAddressList[1].hed__MailingCountry__c,
+            'The country should be ' + newCountry
+        );
+        System.assertEquals(
+            postalCode,
+            newAccountAddressList[1].hed__MailingPostalCode__c,
+            'The postal code should be ' + postalCode
+        );
+        System.assertEquals(
+            newAccountAddressList[1].Id,
+            insertedAccount.hed__Current_Address__c,
+            'The current account address is incorrect'
+        );
+    }
+
+    /**
+     * @description T-3904089 When Deleting an address, the filter field and usernames to exclude filters must work in
+     * the trigger handler for the address trigger ADDR_Addresses_TDTM, the contact mailing address must remain filled in
+     */
+    @isTest
+    private static void deleteAddressRecordLinkedToContactFieldFilterEnabled() {
+        hed__Hierarchy_Settings__c orgSettings = createSettings();
+        TDTM_Utility.disableAddressTriggers();
+        Account newAccount = createTestAccountWithAddress('Administrative');
+        Contact newContact = createTestContactWithAddress(newAccount.Id);
+        TDTM_Utility.enableAddressTriggers();
+
+        //Retrieves the defaultTDTMConfig
+        List<hed.TDTM_Global_API.TdtmToken> tokens = hed.TDTM_Global_API.getDefaultTdtmConfig();
+
+        //Add filter on ADDR_Addresses_TDTM triggerHandler
+        for (hed.TDTM_Global_API.TdtmToken token : tokens) {
```

Comment:
@salvadorsfdc, what is the purpose of this filter token?

Reply:
@iskander-sfdc by providing a filterFIeld and filterValue we can tell the TDTM trigger to ignore executing the logic for the record that matches that field and value criteria. In this case, the engine ignores to run for the record where the MailingStreet__c equals to 'street'

Reply:
Oh, it was in the prerequisite, got it